mmv1/third_party/terraform/acctest/vcr_utils.go

```diff
@@ -242,14 +264,34 @@ func initializeReleaseDiffTest(c resource.TestCase, testName string) resource.Te
 			},
 		}
 	}
+	// InsertDiffSteps adds modified steps to the test that run with an external provider
+	// these steps do the actual infrastructure provisioning, and c.Steps is updated to have these modified steps
+	c.Steps = InsertDiffSteps(c, tempOutputFile, releaseProvider, localProviderName)
+
+	return
+}
+
+// InsertDiffSteps inserts a new step into the test case that reformats the config to use the release provider - this allows us to see the diff
+// between the local provider and the release provider
+func InsertDiffSteps(c resource.TestCase, tempOutputFile *os.File, releaseProvider string, localProviderName string) []resource.TestStep {
+	var countSteps = 0
 
 	var replacementSteps []resource.TestStep
 	for _, testStep := range c.Steps {
+		countSteps++
 		if testStep.Config != "" {
 			ogConfig := testStep.Config
+			fmt.Fprintf(tempOutputFile, "[DEBUG] Original config: %s\n", ogConfig)
 			testStep.Config = ReformConfigWithProvider(ogConfig, localProviderName)
-			if testStep.ExpectError == nil && testStep.PlanOnly == false {
+			fmt.Fprintf(tempOutputFile, "[DEBUG] Reformatted config: %s\n", testStep.Config)
+			testStep.PreConfig = func() {
+				fmt.Fprintf(tempOutputFile, "[Diff] Step %d\n", countSteps)
```

Comment:
Would it make sense to keep `[Diff]` in a constant at the top, since it is used again later during parse? Mainly to prevent accidental changes to the logic that is inherent in these print lines.